### PR TITLE
GitHub Action CI workflow and ArgoCD app setup to local namespace

### PR DESCRIPTION
**Pull request by MarkoKLI**

### CODE REVIEW

File: argocd/local.argocd.yaml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+spec:
+  project: default
+
+  destination:
+    namespace: local
+    server: https://kubernetes.default.svc
+
+  source:
+    repoURL: https://github.com/theelasticband/devops/
+    targetRevision: klincharov
```
Comment (MarkoKLI) [inline argocd/local.argocd.yaml:17]: targetRevision is set to branch klincharov for local testing of branch, for the sake of completeness. 
Value should be changed to HEAD before merge to main.

File: .github/workflows/ci-app1.yaml [hunk 2]
```diff
@@ -6,6 +6,7 @@ on:
       - 'main'
```
Comment (theelasticband) [inline .github/workflows/ci-app1.yaml:7]: add your own branch to test

File: .github/workflows/ci-app2.yaml [hunk 3]
```diff
@@ -19,16 +20,42 @@ jobs:
...
+          
+          if [ -z "${semver[0]}" ] || [ -z "${semver[1]}" ] || [ -z "${semver[2]}" ]; then # basic format check
+            echo "Error: Invalid semver format."
+            exit 1
+          fi
+          
+          echo "GITHUB_SHA_VALUE=`echo $GITHUB_SHA | cut -c 9- `" >> $GITHUB_ENV # setting patch semver version value
+          echo "V_MAJOR=${semver[0]}" >> $GITHUB_ENV # setting major semver version value 
+          echo "V_MINOR=${semver[1]}" >> $GITHUB_ENV # setting minor semver version value
+          echo "V_PATCH=${semver[2]}" >> $GITHUB_ENV # setting patch semver version value
```
Comment (theelasticband) [inline .github/workflows/ci-app2.yaml]: make this auto increase

File: .github/workflows/ci-app1.yaml (hunk 2 above)
Comment (MarkoKLI) [reply .github/workflows/ci-app1.yaml:7]: klincharov branch added in both ci-app1.yaml and ci-app2.yaml

File: .github/workflows/ci-app2.yaml (hunk 3 above)
Comment (MarkoKLI) [reply .github/workflows/ci-app2.yaml]: Patch version number made to change based on GitHub action run number.
Also added a step to write down newest semver in .version file.
Major and minor versions to be changed manually via file.

**Review by theelasticband — APPROVED** [review]

**Pull request merged**